launcher/launcher/main.go

```diff
@@ -171,18 +193,26 @@ func startLauncher(ctx context.Context, launchSpec spec.LaunchSpec, serialConsol
 	}
 	gceAk.Close()
 
-	token, err := registryauth.RetrieveAuthToken(ctx, mdsClient)
+	token, err := registryauth.RetrieveAuthToken(context.Background(), mdsClient)
+	if err != nil {
+		logger.Info(fmt.Sprintf("failed to retrieve auth token: %v, using empty auth for image pulling\n", err))
+	}
+
+	uptime, err := getUptime()
 	if err != nil {
-		logger.Printf("failed to retrieve auth token: %v, using empty auth for image pulling\n", err)
+		logger.Error(fmt.Sprintf("error reading VM uptime: %v", err))
 	}
+	logger.Info("Launch completed", "latency_sec", uptime)
 
-	ctx = namespaces.WithNamespace(ctx, namespaces.Default)
+	ctx := namespaces.WithNamespace(context.Background(), namespaces.Default)
 	r, err := launcher.NewRunner(ctx, containerdClient, token, launchSpec, mdsClient, tpm, logger, serialConsole)
 	if err != nil {
 		return err
 	}
 	defer r.Close(ctx)
 
+	// Start tracking time for workload execution.
+	start = time.Now()
```

Comment:
Technically this is not where the workload actually starts (in `r.Run()`, it will still spend time measuring eventlog, creating attestation etc... before start the workload task), but the measurement is still useful here

Reply:
Got it, is it worth adding another timer when the workload actually starts and/or relabeling this to be more accurate?

Reply:
Looking more into the code, I think we can either:

1. Keep the logs as is (so we're including some setup steps in the "workload duration"), which is slightly inaccurate but simple
2. Include the setup steps in `r.Run()` (event log measurements, attestation) in the launcher duration log (so we log the launcher's duration in `r.Run()` instead of `startLauncher()` - biggest pain here is we'll need to provide the `main()` start time to the container runner (or the logging library) to calculate the duration, which is doable but a bit clumsy.
3. Separately time/log the duration for the `r.Run()` setup, and start timing the workload when it actually starts (after `task.Wait()` returns successfully). This adds yet another metric/field we have to filter for, so I'd only prefer to do this if it's worth monitoring as its own step ("workload setup"?).

I'm generally in favor of 1 or 3 for ease of implementation, but lmk what you think

Reply:
I think 3 makes sense. The setup itself is also worth measuring.